File: SOCIETY/Features/Profile/UserProfile.swift
//
//  UserProfile.swift
//  SOCIETY
//
//  Created by Dino Hukanovic on 31/01/2026.
//

import Foundation

/// Domain model for the current user's profile (identity, contact, and social presence).
struct UserProfile: Equatable {
    var id: UUID
    var firstName: String
    var lastName: String
    var bio: String?
    var username: String
    var email: String
    var phoneNumber: String?
    var profileImageURL: String?

    // Social handles (stored as handles or URLs as appropriate)
    var instagramHandle: String?
    var twitterHandle: String?
    var youtubeHandle: String?
    var tiktokHandle: String?
    var linkedinHandle: String?
    var websiteURL: String?

    /// Display name built from first and last name.
    var fullName: String {
        [firstName, lastName]
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
            .filter { !$0.isEmpty }
            .joined(separator: " ")
            .trimmingCharacters(in: .whitespaces)
    }

    /// For backward compatibility when only a single name is stored (e.g. auth metadata).
    static func fullName(from singleName: String?) -> String {
        singleName?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
    }
}

File: models/UserProfile.ts
export type ProfileRole = "user" | "admin";
export type AvatarSource = "dicebear" | "upload";

export interface UserProfile {
  id: string;
  full_name: string | null;
  avatar_url: string | null;
  first_name: string | null;
  last_name: string | null;
  bio: string | null;
  username: string | null;
  phone_number: string | null;
  instagram_handle: string | null;
  twitter_handle: string | null;
  youtube_handle: string | null;
  tiktok_handle: string | null;
  linkedin_handle: string | null;
  website_url: string | null;
  birthday: string | null;
  avatar_source: AvatarSource | null;
  avatar_seed: string | null;
  avatar_style: string | null;
  onboarding_completed: boolean;
  is_active: boolean;
  deleted_at: string | null;
  last_seen_at: string | null;
  last_app_open_at: string | null;
  expo_push_token: string | null;
  role: ProfileRole;
  created_at: string;
  updated_at: string;
}
